ldap/servers/plugins/roles/roles_cache.c

```diff
@@ -1098,7 +1098,7 @@ roles_cache_create_object_from_entry(Slapi_Entry *role_entry, role_object **resu
     /* We determine the role type by reading the objectclass */
     if (roles_cache_is_role_entry(role_entry) == 0) {
         /* Bad type */
-        slapi_ch_free((void **)&this_role);
+        roles_cache_role_object_free(this_role);
```

Comment:
You need to cast these to avoid compiler warnings:  

    roles_cache_role_object_free( (caddr_t)this_role));

Reply:
Yeah, didn't catch these on 389-ds-base-2.5 branch.